### PR TITLE
Make --serialized-file argument optional

### PR DESCRIPTION
**Pull request by lxning**
## Description

Please include a summary of the feature or issue being fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)
#993 
## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the tests [UT/IT] that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced. 
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B
1.  archive:
torch-model-archiver --model-name densenet161 --version 1.0 --model-file examples/image_classifier/densenet_161/model.py  --extra-files examples/image_classifier/index_to_name.json --handler image_classifier

2.  torchserve  --model-store /home/model-server/model-store --models densenet161.mar

3. curl http://127.0.0.1:8080/predictions/densenet161 -T image_classifier/kitten.jpg
{
  "ashcan": 0.003682350507006049,
  "maraca": 0.003106366842985153,
  "groenendael": 0.0025914516299962997,
  "reflex_camera": 0.002574876882135868,
  "macaw": 0.0025670137256383896
}

- UT/IT execution results

- Logs

## Checklist:

- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] New and existing unit tests pass locally with these changes?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Comment by dhanainme** [comment]
Looks good to me. Can you please confirm the behavior of Torchserve upon testing. 

Were you able to create a MAR file with serialized file.

File: ts/torch_handler/base_handler.py [hunk 1]
```diff
@@ -67,6 +66,9 @@ def initialize(self, context):
             self.model = self._load_pickled_model(model_dir, model_file, model_pt_path)
         else:
             logger.debug("Loading torchscript model")
+            if not os.path.isfile(model_pt_path):
```
Comment (dhanainme) [inline ts/torch_handler/base_handler.py:69]: Looks like presence / absence of modelFile arg in the manifest determines if this is an eager mode / torchscript model. Given usage of this in AWS Inf tolkit 

```
    model_archiver_cmd = [
        "torch-model-archiver",
        "--model-name",
        DEFAULT_TS_MODEL_NAME,
        "--handler",
        handler_service,
        "--serialized-file",
        os.path.join(environment.model_dir, DEFAULT_TS_MODEL_SERIALIZED_FILE),
        "--export-path",
        DEFAULT_TS_MODEL_DIRECTORY,
        "--extra-files",
        os.path.join(environment.model_dir, DEFAULT_TS_CODE_DIR, environment.Environment().module_name + ".py"),
        "--version",
        "1",
    ]

```

https://github.com/aws/sagemaker-pytorch-inference-toolkit/blob/6936c08581e26ff3bac26824b1e4946ec68ffc85/src/sagemaker_pytorch_serving_container/torchserve.py#L110

Does AWS Inf toolkit even support Torchscript models ?

Comment (lxning) [reply ts/torch_handler/base_handler.py:69]: toolkit missed option "--model-file " for eager mode. It should be fixed in toolkit.

Comment (maaquib) [reply ts/torch_handler/base_handler.py:69]: If customers have been using TS archiver without `--model-file` args, we should make sure this doesn't cause any backward compatibility issues

Comment (lxning) [reply ts/torch_handler/base_handler.py:69]: On TS side, the only change for option "--serialized-file" is that it is optional in eager mode. So it is backward compatible.

**Review by dhanainme — APPROVED** [review]

**Review by maaquib — APPROVED** [review]

**Pull request merged**